packages/xstate-solid/src/index.ts

```diff
@@ -0,0 +1,5 @@
+export { useActor } from './useActor';
+export { useInterpret } from './useInterpret';
+export { useMachine, UseMachineReturn } from './useMachine';
```

Comment:
Exporting that to support typing useMachine in Solid Context. Don't need to worry about needless rerenders, so you can pass useMachine in Context rather than an interpreter.

Reply:
Could you share a code snippet showcasing that?

Reply:
Yeah, I actually wrote it up in the recipes page for solid: https://github.com/statelyai/xstate/blob/f22fca2eaee7f946873fc86e36cfcf507f88de60/docs/recipes/solid.md#context-provider

Reply:
I guess we could export it from other packages too - but I sort of think about the return type of this as just a temporary tuple that is supposed to be immediately destructured. So I'm not super keen on making it easier to pass around this tuple as a whole.

I see how this would be useful in the context of this recipe but I left [a comment there](https://github.com/statelyai/xstate/pull/2932/files#r944757781) that potentially could simplify this quite a bit. With that proposal, you wouldn't need this type helper at all because you could simply pass around `InterpreterFrom<typeof machine>`.

Also, note that with TS 4.7+ this can be somewhat easily achieved without custom types like this: `ReturnType<typeof useMachine<typeof machine>>`